player/player.gd

```diff
@@ -113,3 +118,17 @@ func _handle_player_hit(amount: int) -> void:
 
 func _handle_fire_rate_upgrade(level: int):
 	fire_rate =  base_fire_rate - (level * base_fire_rate * 0.1)
+	
+func _handle_gunman_upgrade(level: int):
+	var gunman = Gunman.instantiate()
+	var posts = get_tree().get_nodes_in_group("gunman_posts") as Array[Node2D]
+	var min_occupancy = posts.map(
+		func (p): return p.get_child_count()).reduce(
+			func (min, value): return value if value < min else min)
+	
+	posts.filter(
+		func (p): return p.get_child_count() <= min_occupancy).pick_random().add_child(gunman)
```

Comment:
nice!

Only comment on this would be that it would probably be best if we not include this on the player, could be their own standalone thing, particularly considering that we'll have either more types, or other upgrades that improve them, maybe a "garrison" node?